xpk.py

```diff
@@ -1763,6 +1829,43 @@ def check_if_workload_exists(args) -> bool:
   return False
 
 
+def check_if_workload_can_schedule(args, system):
+  """Check if workload can schedule based on the cluster resources (tpu_type and maximu VM in cluster).
+
+  Args:
+    args: user provided arguments for running the command.
+    system: system characteristics
+
+  Returns:
+    returns true if workload can schedule, otherwise returns false.
+  """
+  cluster_config_map = get_cluster_configmap(args)
+
+  # Prevents workload creation failure for existing clusters with no ConfigMap
+  if cluster_config_map is None:
+    xpk_print(f'No ConfigMap exist for cluster with the name {args.cluster}-configmap.')
+    return True
+
+  if args.tpu_type not in cluster_config_map:
+    xpk_print(f'{args.workload} is requesting {args.tpu_type} but '
+      f'cluster only contains {cluster_config_map.keys()}. '
+      'XPK will not create this workload.'
+    )
+    return False
+
+  max_vm_in_cluster = cluster_config_map[args.tpu_type]
+  vm_required_by_workload = int(args.num_slices) * system.vms_per_slice
+  if vm_required_by_workload > max_vm_in_cluster:
+    xpk_print(
+        f'{args.workload} is requesting {vm_required_by_workload} VMs of {args.tpu_type} '
+        f'but the cluster only contains {max_vm_in_cluster} VMs of {args.tpu_type}. '
```

Comment:
Is it possible to add the number of slices of the tpu type being requested as well? I think it is easier to understand the workload based on the same args the user provides which is in num-slices and tpu type. The vm count is also great since kueue using vm count so having both will be super clear!